lib/core/node_switching_service.dart

```diff
@@ -15,12 +16,16 @@ class NodeSwitchingService {
 
   static const int _healthCheckIntervalSeconds = 30;
 
-  // The number of times we want to reset the overall used trusted nodes list.
-  // We don't want an infinite loop if all trusted nodes are down.
-  static const int _usedTrustedNodeResetCount = 1;
+  // Maximum number of node switching attempts per session
+  static const int _maxNodeSwitchingAttempts = 5;
 
-  // State to manage the reset count
-  int _resetCount = 0;
+  // Cooldown period between node switching attempts (in seconds)
+  static const int _nodeSwitchingCooldownSeconds = 60;
```

Comment:
15 seconds is reasonable enough, since we have the timeout of node connection to 10 seconds anyway
```suggestion
  static const int _nodeSwitchingCooldownSeconds = 15;
```

Reply:
Alright, adjusting to 15 seconds